### PR TITLE
Token type renaming

### PR DESCRIPTION
**Pull request by jackcmay**
TODO: update js bindings

### CODE REVIEW

**Comment by jackcmay** [comment]
@mvines inline with what you were thinking too?

File: token/src/instruction.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ pub struct TokenInfo {
 #[repr(C)]
 #[derive(Clone, Debug, PartialEq)]
 pub enum TokenInstruction {
-    /// Creates a new token and deposit all the newly minted tokens in an account.
+    /// Initializes a new token and deposit all the newly minted tokens in an account.
```
Comment (CriesofCarrots) [inline token/src/instruction.rs:25]:
```suggestion
    /// Initializes a new token and deposits all the newly minted tokens in an account.
```

**Comment by mvines** [comment]
Nice.  How about going a little further where:
1. state::TokenMint becomes just Mint
2. state::Account now becomes Token.

**Comment by jackcmay** [comment]
@mvines Looks like you are referring to the generated C file, what do you think about the Rust naming?

**Comment by jackcmay** [comment]
Could rename `State::Account(Account)` to `State::Account(Token)` but that doesn't seem better

**Comment by mvines** [comment]
Maybe this?
```rust
pub enum State {
    /// Unallocated state, may be initialized into another state.
    Unallocated,
    /// A token type and associated mint.
    TokenType(Mint),
    /// Some amount of tokens
    Token(TokenInfo),
    /// Invalid state, cannot be modified by the token program.
    Invalid,
}
```

I also kinda like how the ERC20 standard has a token name and token symbol strings in it (https://eips.ethereum.org/EIPS/eip-20).  There's certainly room in the `State::TokenType` enum for those strings, since the `Mint` struct is much smaller than the `TokenInfo`.   Name/symbol a little toyish, but it does allow for a generic explorer interface to display all known tokens -- which I think people in the space tend to like.

**Comment by jackcmay** [comment]
@mvines @CriesofCarrots Post commit reviews welcome :-)

**Pull request merged**